diff --git a/src/main/java/net/wurstclient/serverfinder/api/ServerInfoRequest.java b/src/main/java/net/wurstclient/serverfinder/api/ServerInfoRequest.java
new file mode 100644
index 00000000..ddd614ec
--- /dev/null
+++ b/src/main/java/net/wurstclient/serverfinder/api/ServerInfoRequest.java
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2014-2025 Wurst-Imperium and contributors.
+ *
+ * This source code is subject to the terms of the GNU General Public
+ * License, version 3. If a copy of the GPL was not distributed with this
+ * file, You can obtain one at: https://www.gnu.org/licenses/gpl-3.0.txt
+ */
+package net.wurstclient.serverfinder.api;
+
+public class ServerInfoRequest
+{
+	private final String api_key;
+	private final String ip;
+	private final int port;
+	
+	public ServerInfoRequest(String apiKey, String ip, int port)
+	{
+		this.api_key = apiKey;
+		this.ip = ip;
+		this.port = port;
+	}
+}