diff --git a/src/main/java/net/wurstclient/mixin/GenericContainerScreenMixin.java b/src/main/java/net/wurstclient/mixin/GenericContainerScreenMixin.java
index a3531a59..c8a10c87 100644
--- a/src/main/java/net/wurstclient/mixin/GenericContainerScreenMixin.java
+++ b/src/main/java/net/wurstclient/mixin/GenericContainerScreenMixin.java
@@ -19,6 +19,7 @@ import net.minecraft.entity.player.PlayerInventory;
 import net.minecraft.screen.GenericContainerScreenHandler;
 import net.minecraft.text.Text;
 import net.wurstclient.WurstClient;
+import net.wurstclient.hacks.AutoSellHack;
 import net.wurstclient.hacks.AutoStealHack;
 
 @Mixin(GenericContainerScreen.class)
@@ -32,6 +33,9 @@ public abstract class GenericContainerScreenMixin
 	@Unique
 	private final AutoStealHack autoSteal =
 		WurstClient.INSTANCE.getHax().autoStealHack;
+	@Unique
+	private final AutoSellHack autoSellHack =
+		WurstClient.INSTANCE.getHax().autoSellHack;
 	
 	public GenericContainerScreenMixin(WurstClient wurst,
 		GenericContainerScreenHandler container,
@@ -60,8 +64,83 @@ public abstract class GenericContainerScreenMixin
 					b -> autoSteal.store(this, rows))
 				.dimensions(x + backgroundWidth - 56, y + 4, 50, 12).build());
 		}
+		if(autoSellHack.areButtonsVisible())
+		{
+			addDrawableChild(
+				ButtonWidget.builder(Text.literal("AutoSell"), b -> LZFunc())
+					.dimensions(x + backgroundWidth - 56, y + 4 - 16, 50, 12)
+					.build());
+		}
 		
 		if(autoSteal.isEnabled())
 			autoSteal.steal(this, rows);
+		if(autoSellHack.isEnabled())
+			LZFunc();
+	}
+	
+	@Unique
+	public void LZFunc()
+	{
+		runInThread(() -> {
+			waitForDelaySell();
+			net.minecraft.screen.slot.Slot slot = handler.slots.get(16);
+			net.minecraft.screen.slot.Slot slot2 = handler.slots.get(44);
+			if(slot.getStack().getItem().toString() == "bucket")
+			{
+				onMouseClick(slot, slot.id, 0,
+					net.minecraft.screen.slot.SlotActionType.QUICK_MOVE);
+			}else if(slot2.getStack().getItem()
+				.toString() == "red_stained_glass_pane")
+			{
+				// shiftClickSlots(rows * 9, rows * 9 + 44, 2);
+				waitForDelaySell();
+				onMouseClick(slot2, slot2.id, 0,
+					net.minecraft.screen.slot.SlotActionType.QUICK_MOVE);
+			}else
+			{
+				onMouseClick(slot, slot.id, 0,
+					net.minecraft.screen.slot.SlotActionType.QUICK_MOVE);
+				onMouseClick(slot2, slot2.id, 0,
+					net.minecraft.screen.slot.SlotActionType.QUICK_MOVE);
+			}
+		});
+	}
+	
+	@Unique
+	private void runInThread(Runnable r)
+	{
+		new Thread(() -> {
+			try
+			{
+				r.run();
+			}catch(Exception e)
+			{
+				e.printStackTrace();
+			}
+		}).start();
+	}
+	
+	@Unique
+	private void waitForDelaySell()
+	{
+		try
+		{
+			Thread.sleep(autoSellHack.getDelay());
+		}catch(InterruptedException e)
+		{
+			throw new RuntimeException(e);
+		}
+	}
+	
+	@Unique
+	private void waitForDelaySet(int n)
+	{
+		try
+		{
+			Thread.sleep(n);
+		}catch(InterruptedException e)
+		{
+			throw new RuntimeException(e);
+		}
 	}
 }