Index: src/main/java/net/wurstclient/other_features/WurstLogoOtf.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/net/wurstclient/other_features/WurstLogoOtf.java b/src/main/java/net/wurstclient/other_features/WurstLogoOtf.java
--- a/src/main/java/net/wurstclient/other_features/WurstLogoOtf.java	(revision 96786bbce9ac3a2b5a8294ee2079ead6acfd271f)
+++ b/src/main/java/net/wurstclient/other_features/WurstLogoOtf.java	(date 1736021040143)
@@ -33,7 +33,7 @@
 	
 	public WurstLogoOtf()
 	{
-		super("WurstLogo", "Shows the Wurst logo and version on the screen.");
+		super("KoboshLogo", "Shows the Kobosh logo and version on the screen.");
 		addSetting(bgColor);
 		addSetting(txtColor);
 		addSetting(visibility);
