diff --git a/src/main/java/net/wurstclient/serverfinder/api/ServersRequest.java b/src/main/java/net/wurstclient/serverfinder/api/ServersRequest.java
new file mode 100644
index 00000000..89dfba96
--- /dev/null
+++ b/src/main/java/net/wurstclient/serverfinder/api/ServersRequest.java
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2014-2025 Wurst-Imperium and contributors.
+ *
+ * This source code is subject to the terms of the GNU General Public
+ * License, version 3. If a copy of the GPL was not distributed with this
+ * file, You can obtain one at: https://www.gnu.org/licenses/gpl-3.0.txt
+ */
+package net.wurstclient.serverfinder.api;
+
+import com.google.gson.JsonArray;
+
+public class ServersRequest
+{
+	private final String api_key;
+	private Integer asn;
+	private String country_code;
+	private Boolean cracked;
+	private String description;
+	private JsonArray max_players;
+	private Integer online_after;
+	private JsonArray online_players;
+	private Integer protocol;
+	private Boolean ignore_modded;
+	private Boolean only_bungeespoofable;
+	private String software;
+	
+	public enum Software
+	{
+		Any,
+		Bukkit,
+		Spigot,
+		Paper,
+		Vanilla
+	}
+	
+	public ServersRequest(String apiKey)
+	{
+		this.api_key = apiKey;
+	}
+	
+	public void setAsn(Integer asn)
+	{
+		this.asn = asn;
+	}
+	
+	public void setCountryCode(String countryCode)
+	{
+		this.country_code = countryCode;
+	}
+	
+	public void setCracked(Boolean cracked)
+	{
+		this.cracked = cracked;
+	}
+	
+	public void setDescription(String description)
+	{
+		this.description = description;
+	}
+	
+	public void setMaxPlayers(Integer exact)
+	{
+		this.max_players = new JsonArray();
+		this.max_players.add(exact);
+		this.max_players.add(exact);
+	}
+	
+	public void setMaxPlayers(Integer min, Integer max)
+	{
+		this.max_players = new JsonArray();
+		this.max_players.add(min);
+		if(max == -1)
+			this.max_players.add("inf");
+		else
+			this.max_players.add(max);
+	}
+	
+	public void setOnlineAfter(Integer unixTimestamp)
+	{
+		this.online_after = unixTimestamp;
+	}
+	
+	public void setOnlinePlayers(Integer exact)
+	{
+		this.online_players = new JsonArray();
+		this.online_players.add(exact);
+		this.online_players.add(exact);
+	}
+	
+	public void setOnlinePlayers(Integer min, Integer max)
+	{
+		this.online_players = new JsonArray();
+		this.online_players.add(min);
+		if(max == -1)
+			this.online_players.add("inf");
+		else
+			this.online_players.add(max);
+	}
+	
+	public void setProtocolVersion(Integer version)
+	{
+		this.protocol = version;
+	}
+	
+	public void setSoftware(Software software)
+	{
+		this.software = software == Software.Any ? null : software.name();
+	}
+	
+	public void setIgnoreModded(Boolean ignore)
+	{
+		this.ignore_modded = ignore;
+	}
+	
+	public void setOnlyBungeeSpoofable(Boolean only)
+	{
+		this.only_bungeespoofable = only;
+	}
+}