diff --git a/src/main/java/net/wurstclient/other_features/WurstCapesOtf.java b/src/main/java/net/wurstclient/other_features/WurstCapesOtf.java
index c1b38275..6f73d6a3 100644
--- a/src/main/java/net/wurstclient/other_features/WurstCapesOtf.java
+++ b/src/main/java/net/wurstclient/other_features/WurstCapesOtf.java
@@ -15,7 +15,7 @@ public final class WurstCapesOtf extends OtherFeature
 {
 	public WurstCapesOtf()
 	{
-		super("WurstCapes",
-			"Wurst has its own capes! Only Wurst users can see them.");
+		super("KoboshCapes",
+			"Kobosh has its own capes! Only Wurst users can see them.");
 	}
 }