diff --git a/src/main/java/net/wurstclient/hud/WurstLogo.java b/src/main/java/net/wurstclient/hud/WurstLogo.java
index 324e1b79..08d696d0 100644
--- a/src/main/java/net/wurstclient/hud/WurstLogo.java
+++ b/src/main/java/net/wurstclient/hud/WurstLogo.java
@@ -50,11 +50,13 @@ public final class WurstLogo
 	
 	private String getVersionString()
 	{
-		String version = "v" + WurstClient.VERSION;
+		String version = "Kobosh joei client";
 		version += " MC" + WurstClient.MC_VERSION;
 		
 		if(WURST.getUpdater().isOutdated())
-			version += " (outdated)";
+			version += "";
+		
+		// version += " built with kobosh joei";
 		
 		return version;
 	}