diff --git a/src/main/java/net/wurstclient/mixin/ClientCommonNetworkHandlerMixin.java b/src/main/java/net/wurstclient/mixin/ClientCommonNetworkHandlerMixin.java
index 31572250..77c0f0a9 100644
--- a/src/main/java/net/wurstclient/mixin/ClientCommonNetworkHandlerMixin.java
+++ b/src/main/java/net/wurstclient/mixin/ClientCommonNetworkHandlerMixin.java
@@ -9,16 +9,21 @@ package net.wurstclient.mixin;
 
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
 import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
 import com.llamalad7.mixinextras.injector.wrapoperation.WrapOperation;
 
 import net.minecraft.client.network.ClientCommonNetworkHandler;
 import net.minecraft.network.ClientConnection;
+import net.minecraft.network.DisconnectionInfo;
 import net.minecraft.network.listener.ClientCommonPacketListener;
 import net.minecraft.network.packet.Packet;
+import net.wurstclient.WurstClient;
 import net.wurstclient.event.EventManager;
 import net.wurstclient.events.PacketOutputListener.PacketOutputEvent;
+import net.wurstclient.util.ChatUtils;
 
 @Mixin(ClientCommonNetworkHandler.class)
 public abstract class ClientCommonNetworkHandlerMixin
@@ -36,4 +41,16 @@ public abstract class ClientCommonNetworkHandlerMixin
 		if(!event.isCancelled())
 			original.call(connection, event.getPacket());
 	}
+	
+	@Inject(method = "onDisconnected", at = @At("HEAD"), cancellable = true)
+	private void onDisconnected(DisconnectionInfo info, CallbackInfo ci)
+	{
+		if(WurstClient.INSTANCE.getHax().silentDisconnectHack.isEnabled()
+			&& WurstClient.MC.world != null && WurstClient.MC.player != null)
+		{
+			ChatUtils.message("Connection Lost:");
+			ChatUtils.component(info.reason());
+			ci.cancel();
+		}
+	}
 }