diff --git a/src/main/java/net/wurstclient/options/WurstOptionsScreen.java b/src/main/java/net/wurstclient/options/WurstOptionsScreen.java
index 9e989032..fdfa7801 100644
--- a/src/main/java/net/wurstclient/options/WurstOptionsScreen.java
+++ b/src/main/java/net/wurstclient/options/WurstOptionsScreen.java
@@ -120,26 +120,31 @@ public class WurstOptionsScreen extends Screen
 	{
 		OperatingSystem os = Util.getOperatingSystem();
 		
-		new WurstOptionsButton(54, 24, () -> "Official Website",
-			"§n§lWurstClient.net",
-			b -> os.open("https://www.wurstclient.net/options-website/"));
-		
-		new WurstOptionsButton(54, 48, () -> "Wurst Wiki", "§n§lWurst.Wiki",
-			b -> os.open("https://www.wurstclient.net/options-wiki/"));
-		
-		new WurstOptionsButton(54, 72, () -> "WurstForum", "§n§lWurstForum.net",
-			b -> os.open("https://www.wurstclient.net/options-forum/"));
-		
-		new WurstOptionsButton(54, 96, () -> "Twitter", "@Wurst_Imperium",
-			b -> os.open("https://www.wurstclient.net/options-twitter/"));
-		
-		new WurstOptionsButton(54, 120, () -> "Donate",
-			"§n§lWurstClient.net/donate\n"
-				+ "Donate now to help me keep the Wurst Client alive and free"
-				+ " to use for everyone.\n\n"
-				+ "Every bit helps and is much appreciated! You can also get a"
-				+ " few cool perks in return.",
-			b -> os.open("https://www.wurstclient.net/options-donate/"));
+		new WurstOptionsButton(54, 24, () -> "Kobosh Website", "§n§lkobosh.com",
+			b -> os.open("https://kobosh.com"));
+		//
+		// new WurstOptionsButton(54, 48, () -> " Wiki", "§n§lWurst.Wiki\n"
+		// + "We are looking for volunteers to help us expand"
+		// + " the wiki and keep it up to date with the latest Wurst updates.",
+		// b -> os.open(
+		// "https://wurst.wiki/?utm_source=Wurst+Client&utm_medium=Wurst+Options&utm_content=Wurst+Wiki"));
+		//
+		// new WurstOptionsButton(54, 72, () -> "WurstForum",
+		// "§n§lWurstForum.net",
+		// b -> os.open(
+		// "https://wurstforum.net/?utm_source=Wurst+Client&utm_medium=Wurst+Options&utm_content=WurstForum"));
+		//
+		// new WurstOptionsButton(54, 96, () -> "Twitter", "@Wurst_Imperium",
+		// b -> os.open("https://www.wurstclient.net/twitter/"));
+		//
+		// new WurstOptionsButton(54, 120, () -> "Donate",
+		// "§n§lWurstClient.net/donate\n"
+		// + "Donate now to help me keep the Wurst Client alive and free"
+		// + " to use for everyone.\n\n"
+		// + "Every bit helps and is much appreciated! You can also get a"
+		// + " few cool perks in return.",
+		// b -> os.open(
+		// "https://www.wurstclient.net/donate/?utm_source=Wurst+Client&utm_medium=Wurst+Options&utm_content=Donate"));
 	}
 	
 	@Override
@@ -167,7 +172,7 @@ public class WurstOptionsScreen extends Screen
 		int y1 = 40;
 		int y2 = height / 4 + 24 - 28;
 		
-		context.drawCenteredTextWithShadow(tr, "Wurst Options", middleX, y1,
+		context.drawCenteredTextWithShadow(tr, "Kobosh Options", middleX, y1,
 			Colors.WHITE);
 		
 		context.drawCenteredTextWithShadow(tr, "Settings", middleX - 104, y2,