diff --git a/src/main/java/net/wurstclient/hacks/BookKickHack.java b/src/main/java/net/wurstclient/hacks/BookKickHack.java
new file mode 100644
index 00000000..fd93af6d
--- /dev/null
+++ b/src/main/java/net/wurstclient/hacks/BookKickHack.java
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2014-2025 Wurst-Imperium and contributors.
+ *
+ * This source code is subject to the terms of the GNU General Public
+ * License, version 3. If a copy of the GPL was not distributed with this
+ * file, You can obtain one at: https://www.gnu.org/licenses/gpl-3.0.txt
+ */
+package net.wurstclient.hacks;
+
+import java.util.List;
+import java.util.Optional;
+
+import net.wurstclient.Category;
+import net.wurstclient.SearchTags;
+import net.wurstclient.hack.Hack;
+import net.minecraft.item.Items;
+import net.minecraft.network.packet.c2s.play.BookUpdateC2SPacket;
+import net.wurstclient.util.ChatUtils;
+
+@SearchTags({"dupe", "book"})
+public final class BookKickHack extends Hack
+{
+	public BookKickHack()
+	{
+		super("BookKick");
+		setCategory(Category.ITEMS);
+	}
+	
+	@Override
+	protected void onEnable()
+	{
+		assert MC.player != null;
+		if(!(MC.player.getInventory().getSelectedStack()
+			.getItem() == Items.WRITABLE_BOOK))
+		{
+			ChatUtils.error("Please hold a writable book!");
+			setEnabled(false);
+			return;
+		}
+		MC.player.networkHandler.sendPacket(new BookUpdateC2SPacket(
+			MC.player.getInventory().getSelectedSlot(), List.of(""),
+			Optional.of("The quick brown fox jumps over the lazy dog")));
+		setEnabled(false);
+	}
+}